presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -597,6 +604,14 @@ private PlanBuilder window(PlanBuilder subPlan, QuerySpecification node)
             Expression parametersReplaced = ExpressionTreeRewriter.rewriteWith(new ParameterRewriter(analysis.getParameters(), analysis), windowFunction);
             outputTranslations.addIntermediateMapping(windowFunction, parametersReplaced);
             Expression rewritten = subPlan.rewrite(parametersReplaced);
+
+            // If refers to existing symbol, don't create another PlanNode
+            if (rewritten instanceof SymbolReference) {
```

Comment:
This is an actual fix.
The key thing is to let all functions be processed because some of them may be "decorated" by `cast` (coercions happen inside `subPlan.rewrite`). If a function was already processed before (at this point we have a definitive answer [after coercions processing]), we should skip it and don't create the `PlanNode` though.